src/bm_sim/P4Objects.cpp

```diff
@@ -1888,6 +1947,11 @@ P4Objects::init_pipelines(const Json::Value &cfg_root,
 
           table->set_default_default_entry(action, std::move(adata),
                                            is_action_entry_const);
+        } else {
+          throw json_exception(
+              EFormat() << "'default_entry' of table '" << table_name
+              << "' should have key 'action_data'",
+              cfg_default_entry);
```

Comment:
see comment above. I am not sure we need to enforce this new restriction. But I don't feel very strongly about it, so if you think this is more meaningful this way, we can add it. I assume it won't break the test JSON files included in this repo?

Reply:
All behavioral-model CI tests are passing, as you can see on Github page for this PR.

I have also run all p4c tests with behavioral-model with the changes on this PR, and this exception never occurs.

I am OK removing the check if you prefer.  The main reason I even include it is that it makes it more clear what behavioral-model can rely on going forward, and reduce the number of cases that we need to think about when modifying code related to this.

Reply:
Updated docs in commit 9 to make it clearer that `action_data` key is required inside of `default_entry`.